callbacks.go

```diff
@@ -0,0 +1,605 @@
+		addrCache := s.addrManager.AddressCache()
+
+		// Push the addresses.
+		err := p.PushAddrMsg(addrCache)
+		if err != nil {
+			p.LogError("Can't push address message to %s: %v", p, err)
+			p.Disconnect()
+			return
+		}
+	}
+
+	handleAddrMsg := func(p *peer.Peer, msg *wire.MsgAddr) {
+		// Ignore addresses when running on the simulation test network.  This
+		// helps prevent the network from becoming another public test network
+		// since it will not be able to learn about other peers that have not
+		// specifically been provided.
+		if cfg.SimNet {
+			return
+		}
+
+		// Ignore old style addresses which don't include a timestamp.
+		if p.ProtocolVersion() < wire.NetAddressTimeVersion {
+			return
+		}
+
+		// A message that has no addresses is invalid.
+		if len(msg.AddrList) == 0 {
+			p.LogError("Command [%s] from %s does not contain any addresses",
+				msg.Command(), p)
+			p.Disconnect()
+			return
+		}
+
+		for _, na := range msg.AddrList {
+			// Don't add more address if we're disconnecting.
+			if !p.Connected() {
+				return
+			}
+
+			// Set the timestamp to 5 days ago if it's more than 24 hours
+			// in the future so this address is one of the first to be
+			// removed when space is needed.
+			now := time.Now()
+			if na.Timestamp.After(now.Add(time.Minute * 10)) {
+				na.Timestamp = now.Add(-1 * time.Hour * 24 * 5)
+			}
+
+			// Add address to known addresses for this peer.
+			p.KnownAddresses[addrmgr.NetAddressKey(na)] = struct{}{}
+		}
+
+		// Add addresses to server address manager.  The address manager handles
+		// the details of things such as preventing duplicate addresses, max
+		// addresses, and last seen updates.
+		// XXX bitcoind gives a 2 hour time penalty here, do we want to do the
+		// same?
+		s.addrManager.AddAddresses(msg.AddrList, p.NA())
+	}
+
+	p.AddVersionMsgListener("handleVersionMsg", handleVersionMsg)
+	p.AddMemPoolMsgListener("handleMemPoolMsg", handleMemPoolMsg)
+	p.AddTxMsgListener("handleTxMsg", handleTxMsg)
+	p.AddBlockMsgListener("handleBlockMsg", handleBlockMsg)
+	p.AddInvMsgListener("handleInvMsg", handleInvMsg)
+	p.AddHeadersMsgListener("handleHeadersMsg", handleHeadersMsg)
+	p.AddGetDataMsgListener("handleGetDataMsg", handleGetDataMsg)
+	p.AddGetBlocksMsgListener("handleGetBlocksMsg", handleGetBlocksMsg)
+	p.AddGetHeadersMsgListener("handleGetHeadersMsg", handleGetHeadersMsg)
+	p.AddFilterAddMsgListener("handleFilterAddMsg", handleFilterAddMsg)
+	p.AddFilterClearMsgListener("handleFilterClearMsg", handleFilterClearMsg)
+	p.AddFilterLoadMsgListener("handleFilterLoadMsg", handleFilterLoadMsg)
+	p.AddGetAddrMsgListener("handleGetAddrMsg", handleGetAddrMsg)
+	p.AddAddrMsgListener("handleAddrMsg", handleAddrMsg)
+}
+
+// pushTxMsg sends a tx message for the provided transaction hash to the
+// connected peer.  An error is returned if the transaction hash is not known.
+var pushTxMsg = func(p *peer.Peer, s *server, sha *wire.ShaHash, doneChan, waitChan chan struct{}) error {
```

Comment:
Any reason this is a var instead of just:

```Go
func pushTxMsg(p *peer.Peer, s *server, sha *wire.ShaHash, doneChan, waitChan chan struct{}) error {
```